### PR TITLE
Display plugin lock

### PR DESCRIPTION
**Pull request by sethalves**
- possible fix for a display-plugin related crash

### CODE REVIEW

**Comment by sethalves** [comment]
maybe https://app.asana.com/0/26485730907942/108031117798592

**Comment by hifi-gustavo** [comment]
Build console output for commit: a683360ed4349cba7732815d3e89195aedf075a5 available at: https://gist.github.com/b4b6a611407e7bf8cb81210979007c50<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7570.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7570.dmg

**Comment by sethalves** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: a683360ed4349cba7732815d3e89195aedf075a5 available at: https://gist.github.com/b4b6a611407e7bf8cb81210979007c50<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7570.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/HighFidelity-Alpha-PR7570.dmg

**Comment by cozza13** [comment]
**Testing Notes**

Smoke test this.

**Comment by TNS-TrueNorth** [comment]
Any particular test notes or test plan for this? We ran a brief smoke test of this and nothing in particular stood out or crashed.

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -4789,6 +4789,7 @@ DisplayPlugin* Application::getActiveDisplayPlugin() {
             updateDisplayMode();
             Q_ASSERT(_displayPlugin);
         }
+        std::unique_lock<std::mutex> lock(_displayPluginLock);
         result = _displayPlugin.get();
     } else {
         std::unique_lock<std::mutex> lock(_displayPluginLock);
```
Comment (AndrewMeadows) [inline interface/src/Application.cpp]: I don't expect this change to help.  The code where the lock is being added is already on the main thread (which is what the `if` is verifying at the top of the block) and the `_displayPlugin` can only be updated on the main thread (see the code inside `Application::updateDisplayMode()`) else it is supposed to log a "fatal" message.

That said... `updateDisplayMode()` appears to be connected as a **SLOT** to a menu action which seems like maybe a bad idea...  it isn't clear to me that the slot will only be called on the main thread... but it it weren't we should see a fatal message in the logs... right?

Comment (sethalves) [inline interface/src/Application.cpp]: Yeah, I agree.  closing this one.

**Pull request closed without merging**